owsi-core/owsi-core-components/owsi-core-component-jpa/src/main/java/fr/openwide/core/jpa/batch/executor/BatchRunnable.java

```diff
@@ -0,0 +1,32 @@
+package fr.openwide.core.jpa.batch.executor;
+
+import java.util.List;
+
+public abstract class BatchRunnable<E> {
+
+	public void preExecute(List<Long> allIds) {
+	}
+
+	public void executePartition(List<E> partition) {
+		preExecutePartition(partition);
+
+		for (E entity : partition) {
+			executeUnit(entity);
+		}
+
+		postExecutePartition(partition);
+	}
+
+	public void preExecutePartition(List<E> partition) {
+	}
+
+	public void executeUnit(E unit) {
```

Comment:
Depends on how you override executePartition.

I'm not very enthusiast at the idea of having 2 different implementations.